daq/faucet_states_collector.py

```diff
@@ -66,53 +80,82 @@ def get_switches(self):
             switch_data[switch_name] = self.get_switch(switch_name)
         return switch_data
 
+    def get_switch_map(self):
+        """returns switch map for topology overview"""
+        switch_map = {}
+        topo_obj = self.topo_state
+        with self.lock:
+            for switch in topo_obj.get(TOPOLOGY_GRAPH, {}).get("nodes", []):
+                _id = switch.get("id")
+                if _id:
+                    switch_map[_id] = {}
+                    switch_map[_id]["status"] = None
+        return switch_map
+
     def get_switch(self, switch_name):
         """get switches state"""
         switch_map = {}
-
-        # filling switch attributes
-        attributes_map = switch_map.setdefault("attributes", {})
-        attributes_map["name"] = switch_name
-        attributes_map["dp_id"] = self.switch_states.get(str(switch_name), {}).get(KEY_DP_ID, "")
-        attributes_map["description"] = None
-
-        # filling switch dynamics
-        switch_states = self.switch_states.get(str(switch_name), {})
-        switch_map["config_change_count"] = switch_states.get(KEY_CONFIG_CHANGE_COUNT, "")
-        switch_map["config_change_type"] = switch_states.get(KEY_CONFIG_CHANGE_TYPE, "")
-        switch_map["config_change_timestamp"] = switch_states.get(KEY_CONFIG_CHANGE_TS, "")
-
-        switch_port_map = switch_map.setdefault("ports", {})
-
-        # filling port information
-        for port_id, port_states in switch_states.get(KEY_PORTS, {}).items():
-            port_map = switch_port_map.setdefault(port_id, {})
-            # port attributes
-            switch_port_attributes_map = port_map.setdefault("attributes", {})
-            switch_port_attributes_map["description"] = None
-            switch_port_attributes_map["stack_peer_switch"] = None
-            switch_port_attributes_map["stack_peer_port"] = None
-
-            # port dynamics
-            port_map["status_up"] = port_states.get(KEY_PORT_STATUS_UP, "")
-            port_map["port_type"] = None
-            port_map["status_timestamp"] = port_states.get(KEY_PORT_STATUS_TS, "")
-            port_map["status_count"] = port_states.get(KEY_PORT_STATUS_COUNT, "")
-            port_map["packet_count"] = None
-
-        # filling learned macs
-        switch_learned_mac_map = switch_map.setdefault("learned_macs", {})
-        for mac in switch_states.get(KEY_LEARNED_MACS, set()):
-            mac_map = switch_learned_mac_map.setdefault(mac, {})
-            mac_states = self.learned_macs.get(mac, {})
-            mac_map["ip_address"] = mac_states.get(KEY_MAC_LEARNING_IP, "")
-
-            learned_switch = mac_states.get(KEY_MAC_LEARNING_SWITCH, {}).get(switch_name, {})
-            mac_map["port"] = learned_switch.get(KEY_MAC_LEARNING_PORT, "")
-            mac_map["timestamp"] = learned_switch.get(KEY_MAC_LEARNING_TS, "")
+        with self.lock:
```

Comment:
create a function get_switch_raw() and then call that from get_switch/with -- better than indenting the entire (already too long) function.

def get_switch():
  with self.lock:
    self.get_switch_raw()